### DISCUSSION TITLE
Multiple bot instances

### DISCUSSION THREAD

**Discussion by segman1** [thejoabo/virtualfisher-bot#41] [Ideas]
make it so you can farm in multiple accounts
idk if this is added

**Comment by thejoabo** [comment]
Hello @segman1, it is already added, you can create multiple configs (with different account tokens) and select which config to use when you start up the bot. To run them simultaneously you just need to open another instance of the autofisher on CMD. When it detects multiple configs on the _./configs/_ folder it will prompt you to choose:

![image](https://user-images.githubusercontent.com/49594896/194598768-10c5ac3a-d2b7-4894-8dcc-f6f834f47f79.png)



**Note**: I wouldn't recommend to run multiple instances at the same time unless you proxy/vpn one of the instances. Otherwise you will probably get banned/rate limited by discord's servers.

**Reply by Yurkhun** [reply]
> Hello @segman1, it is already added, you can create multiple configs (with different account tokens) and select which config to use when you start up the bot. To run them simultaneously you just need to open another instance of the autofisher on CMD. When it detects multiple configs on the _./configs/_ folder it will prompt you to choose:
> 
> ![image](https://user-images.githubusercontent.com/49594896/194598768-10c5ac3a-d2b7-4894-8dcc-f6f834f47f79.png)
> 
> **Note**: I wouldn't recommend to run multiple instances at the same time unless you proxy/vpn one of the instances. Otherwise you will probably get banned/rate limited by discord's servers.

would it be possible to use only one config file with multiple user-tokens? to run multiple accounts at the same cmd to control everything in one screen... Opened multiple cmd's to run 10 accounts but always need to change window, it could be a great Idea if in the future we can use multiple accounts at once... in the config might have something like user-token[1]:  .... user-token[2]: .... user-token[3]:  
and in channel_id too (channel_id [1] .... channel_id [2]) .... I don't know how to code so I have no idea how to make it to work....

sorry about my horrible english

**Reply by Yurkhun** [reply]
https://user-images.githubusercontent.com/95500637/207928118-3ab41204-9ed8-43b0-a18b-6274735b8e63.mp4

I did it in a very simple and dirty way cuz I know nothing about coding... but it would be much better if we could have it all in the same config opening in a only one cmd instead of many cmd opening .... if possible 👍 

my english is really bad, sorry about that 😞

**Reply by thejoabo** [reply]
> would it be possible to use only one config file with multiple user-tokens? to run multiple accounts at the same cmd to control everything in one screen... Opened multiple cmd's to run 10 accounts but always need to change window, it could be a great Idea if in the future we can use multiple accounts at once...

Hello @Yurkhun,

That will be hard to do, it would lead to huge logic changes and likely, a ton of errors (also it would be a lot harder to debug and track what causes the errors). The more compartmentalized it is, the better. The best solution is to do exactly what you did in your video, open them in one console with multiple tabs. In the future I might add a "summary" tool to keep track of multiple instances.

**Note**: Be careful, Discord is very annoying when it comes to global limit rates, they can ban all your accounts at once if you reach that limit, I handle the "local" rate limits, so you don't have to worry when running a single instance, but since you are running 10 of them (and they don't communicate with each other to coordinate the message rate), I'm really astonished that you are not banned yet. If you plan to keep doing so, use the proxy functionality (you can easily set them up in your config files), that will keep Discord happy.

_beautiful discord theme btw =)_

**Reply by Yurkhun** [reply]
> > would it be possible to use only one config file with multiple user-tokens? to run multiple accounts at the same cmd to control everything in one screen... Opened multiple cmd's to run 10 accounts but always need to change window, it could be a great Idea if in the future we can use multiple accounts at once...
> 
> Hello @Yurkhun,
> 
> That will be hard to do, it would lead to huge logic changes and likely, a ton of errors (also it would be a lot harder to debug and track what causes the errors). The more compartmentalized it is, the better. The best solution is to do exactly what you did in your video, open them in one console with multiple tabs. In the future I might add a "summary" tool to keep track of multiple instances.
> 
> **Note**: Be careful, Discord is very annoying when it comes to global limit rates, they can ban all your accounts at once if you reach that limit, I handle the "local" rate limits, so you don't have to worry when running a single instance, but since you are running 10 of them (and they don't communicate with each other to coordinate the message rate), I'm really astonished that you are not banned yet. If you plan to keep doing so, use the proxy functionality (you can easily set them up in your config files), that will keep Discord happy.
> 
> _beautiful discord theme btw =)_

Oh thanks for your reply, I was just testing to run it in 10 accounts at the same time, put I guess I will only run 2 - 3 to see if I get ban or not... but thanks for the consideration. maybe I will try running 10 at the same time but each with a different proxy would be good too (But so far I was testing proxies and didn't make any of them to work, IDK why)

The "summary" will be really great implement to the bots

-> the theme is really nice, I love it =)

**Reply by thejoabo** [reply]
All right, I added this function to my to-do list (as I said it might take a while till is implemented). About the proxies, they need to be HTTPS to work properly, I listed 3 proxy indexes in the [Config Guide](https://github.com/thejoabo/autofishbot/blob/main/assets/faq.md#proxy_ip--proxy_port).

**Reply by Yurkhun** [reply]
A suggestion for the proxies, would be cool instead of putting one proxy in each config file... having a proxy.txt file that will be opened automaticly and the bot would connect to the first proxy working in the list...
or it check all the proxies and connects to the best one.. like doing a proxy checker work...


![image](https://user-images.githubusercontent.com/95500637/207937795-9882e2e2-d008-43ed-9ff3-8a4e79259980.png)
for the moment all the proxies I try always lose connection :(

**Reply by thejoabo** [reply]
True, I'll do that in future updates.

**Reply by Yurkhun** [reply]
https://user-images.githubusercontent.com/95500637/207975477-a4b297aa-35c1-4b83-b1ac-2a2f6a1c1adb.mp4



> True, I'll do that in future updates.
Thanks :D


after trying for a while I got 10 working proxies to use in the bots😃 and I tried to improve a little the start up of them, by doing a little menu to make easier to select wich account to launch in a lazy way 😅

**Reply by Yurkhun** [reply]
just to let people know, when I was using all the accounts in the same IP I was getting to /verify all the time but when I used the proxy... I almost don't get the /verify anymore... idk why, it's kinda weird

**Reply by thejoabo** [reply]
That's seems like a great workaround for this issue (while I still don't have a proper solution). Feel free to share the source code here, if you like.

> just to let people know, when I was using all the accounts in the same IP I was getting to /verify all the time but when I used the proxy... I almost don't get the /verify anymore... idk why, it's kinda weird

I'm not so sure about that, I never tested that way (but I'm really not sure if would impact in any way the captchas frequency since they are controlled by the VirtualFisher servers - which don't have any access to user's IPs), if you have some free time, you could let all those accounts run for a while to see how many you get. If I'm not mistaken the frequency is about 1 captcha per 600-700 commands.

Anyways, thanks for your suggestions =)

**Reply by Yurkhun** [reply]
> That's seems like a great workaround for this issue (while I still don't have a proper solution). Feel free to share the source code here, if you like.
> 
> > just to let people know, when I was using all the accounts in the same IP I was getting to /verify all the time but when I used the proxy... I almost don't get the /verify anymore... idk why, it's kinda weird
> 
> I'm not so sure about that, I never tested that way (but I'm really not sure if would impact in any way the captchas frequency since they are controlled by the VirtualFisher servers - which don't have any access to user's IPs), if you have some free time, you could let all those accounts run for a while to see how many you get. If I'm not mistaken the frequency is about 1 captcha per 600-700 commands.
> 
> Anyways, thanks for your suggestions =)

need further tests to see if the IP change anything I dont know but before using the proxy I got 4 accounts simultaniously with the 15 minutes suspension for missing captcha and I got it more frequently, after using the proxy I didnt happen until now... but maybe it's the API (the one who resolves the image to text)  fault when using many in the same IP they fail more maybe that increase the number of captchas I get to resolve 🤔 idk Need more tests on that to comprove

**Reply by Yurkhun** [reply]
I will share the source code, give me some time... I'm in the middle of a valorant match right now 😆

**Reply by Yurkhun** [reply]
[1.txt](https://github.com/thejoabo/autofishbot/files/10241348/1.txt)
[Start.txt](https://github.com/thejoabo/autofishbot/files/10241349/Start.txt)

the start.bat file code is:
https://pastebin.com/LaSFjp9h

in the config folder you need to create a name.cmd  file for each .config you did with this code:
`title (any name you want to give)
python autofishbot.py 1`
for the first .config file the command will be python autofishbot.py 1 for the second it will be python autofishbot.py 2 .... in sequence. I used the title just to identify them later on.

you can create the files as .cmd or .bat it will work the same. I put the code to download .txt if y'all want or just ctrl + c it 😄 

https://user-images.githubusercontent.com/95500637/207987200-4be2457e-9ece-4710-859d-fafe3980deef.mp4

**Reply by Yurkhun** [reply]
Oh in the video the bot does not start because I left blank the user_token ... to show in the video the file ✌️ 
but there is an video from before showing it working .... it's very simple but it works 👍

**Reply by thejoabo** [reply]
Thanks for sharing (I edited your comment and pasted the source code in the pastebin to increase this discussion readability).

**Reply by Yurkhun** [reply]
> Thanks for sharing (I edited your comment and pasted the source code in the pastebin to increase this discussion readability).

oh thanks to you for this amazing bot you made 👍wish you success to you man in life and everything you do  🥇

**Comment by segman1** [comment]
how do you get multiple configs? do you jsut piuy .config int he end?

**Reply by thejoabo** [reply]
You can just copy and paste the preexisting config and change the values corresponding to the other account. 

- All configs must have the ".config" extension
- All configs must be in the "./configs/" folder (relative to your base repository folder)

Also, if you need help to set everything up you can check [FAQ](https://github.com/thejoabo/virtualfisher-bot/blob/main/assets/faq.md) or the [example config file](https://github.com/thejoabo/virtualfisher-bot/blob/main/assets/template.config).

**Reply by segman1** [reply]
okay, very stupid fucking question: how do you access repository folder (im shit at coding)

**Reply by thejoabo** [reply]
The folder where the **autofishbot.py** is located, you should have something like this:
![image](https://user-images.githubusercontent.com/49594896/196238730-63767330-dc2e-40d1-a01a-6268e0f9dddf.png)

**Reply by segman1** [reply]
the problem is i cant locate it

**Reply by thejoabo** [reply]
And how did you ran the bot this whole time ? To run the bot you must execute the autofishbot.py in your terminal. How are you even able to run it if you can't find this folder ? Also, which folder are you trying to locate ? In order to run the bot it automatically creates a config in that directory ((base_folder)/configs/). I'm confused whats causing you so much trouble.

**Reply by segman1** [reply]
the thing is i have to git clone the thing and pip install the req but the config is already set up, i cant just run autofish.py it wont work so

**Reply by segman1** [reply]
if i dont it qill say something along the lines of "autofishbot.py was not found" or soemthing
